core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -118,7 +128,7 @@ abstract class AbstractFetcherManager[T <: AbstractFetcherThread](val name: Stri
   }
 
   // to be defined in subclass to create a specific fetcher
-  def createFetcherThread(fetcherId: Int, sourceBroker: BrokerEndPoint): T
+  def createFetcherThread(fetcherId: Int, sourceBroker: BrokerEndPoint, markPartitionsFailed: Set[TopicPartition] => Unit): T
```

Comment:
Hmm.. Not sure we need to pass this through `createFetcherThread`. Couldn't the `FetcherManager` implementations reference `markPartitionsFailed` directly?

Reply:
Yeah, right, addressed it.